build.gradle

```diff
@@ -62,6 +63,29 @@ spotless {
 
 jar { dependsOn checkLicenses }
 
+class FileStateChecker extends DefaultTask {
+    Set<File> files
+    String knownHash
+
+    @TaskAction
+    def CheckState() {
+        def allContents = this.files.sort()*.readLines()*.join('\n').join('')
+        def currentHash = MessageDigest.getInstance("SHA-256")
+        .digest(allContents.getBytes("UTF-8")).encodeBase64().toString()
```

Comment:
Can you run `splotlessApply` - hopefully it will make this indent better (`spotlessCheck` doesn't raise errors in grade files due to bugs in spotless but apply can reformat them)